components/StreamField/StreamField.tsx

```diff
@@ -0,0 +1,39 @@
+import { SectionBlock, IssuesBlock, NewsBlock, TopicsBlock } from './blocks';
+import type { StreamFieldProps } from './types';
+import styles from './StreamField.module.scss';
+
+const BLOCKS = {
+    section: SectionBlock,
+    issues: IssuesBlock,
+    news: NewsBlock,
+} as const;
+type BlockType = keyof typeof BLOCKS;
```

Comment:
I think it'd be good to declare the types here.

```
type BlockType = 'section' | 'issues' | 'news';
type BLOCKS: Record<BlockType, React.ComponentType> = {
     ...
}
```

I've not tested this so ignore if this doesn't work.

Reply:
I've tried it but I was having type errors so I kept it as is.